vllm/worker/cache_engine.py

```diff
@@ -78,11 +79,18 @@ def _allocate_kv_cache(
             # null block in CpuGpuBlockAllocator requires at least that
             # block to be zeroed-out.
             # We zero-out everything for simplicity.
-            kv_cache.append(
-                torch.zeros(kv_cache_shape,
-                            dtype=self.dtype,
-                            pin_memory=pin_memory,
-                            device=device))
+            if pin_memory:
```

Comment:
Why is this if needed?

Reply:
I have reverted all the changes in this file, as this _allocate_kv_cache() was not used in HPU.